### PR TITLE
Group and reorder the Main Menu (tabs & icons)

### LINKED ISSUE #4666: Main Menu: group & order menu items
mmgrouping

|Description of Issue|Possible Solution|  Warnings |
|:-----------------------------------------:|:-------------------------------------------------:|:---------------:|
| Too many unordered items in the tabs of the Main Menu|Reorder items, so that those with similar meaning or function go together;<br/>Add horizontal lines to group functions that belong together| Only create sensible groups;<br/>do not create more than 3 groups per tab|

### LINKED ISSUE #4667: Main Menu: change icons
mmgrouping

|Description of Issue|Possible Solution|  Warnings |
|:-----------------------------------------:|:-------------------------------------------------:|:---------------:|
| Icons in Main Menu are not recognizable or can be confused with other icons|Change icons or remove them| use icons, that are distinguishable at small sizes|
|some icons are ambiguous|use unique icons with a clear meaning|be careful because icons direct attention to one menu item and away from other items|

### LINKED ISSUE #4668: Main Menu: change File tab
mmfile

|Description of Issue|Possible Solution|  Warnings |
|:-----------------------------------------:|:-------------------------------------------------:|:---------------:|
| separate _new library_ items for LaTeX and Biblatex are used|use hierarchy to reduce number of items in the tab| -|
|_Import_ and _Export_ share one item with one masive submenu| separate export and import for easier use| keep import submenu, but change the export submenu as shown above.|
|-|-|other changes can be found in issues #4665 , #4666 , #4667 |

### LINKED ISSUE #4669: Main Menu: change Edit tab
mmedit

|Description of Issue|Possible Solution|  Warnings |
|:-----------------------------------------:|:-------------------------------------------------:|:---------------:|
| _Rank_ feature rated as unimportant in a user survey| Remove _Rank_ from main menu| -|
|complex editing functions should be tools;<br/>sending as email does not edit anything|move _Send as email_ and _Replace string_ to Tools tab|see prototype in #4672 |
| _Set/clear/rename/append fields_ applies to whole library;<br/>item name is too long;|move to Library tab;<br/>rename to _manage field names&content_| see prototype in #4670 |
|-|-|other changes can be found in issues #4665 , #4666 , #4667 |

### LINKED ISSUE #4670: Main Menu: change Library tab
mmlib

|Description of Issue|Possible Solution|  Warnings |
|:-----------------------------------------:|:-------------------------------------------------:|:---------------:|
| _New article_ shortcut already exists in toolbar|remove _New article_ from Main Menu| -|
| _Edit strings_ is an imprecise function name|find a better name (e.g. see prototype)| do not use overly long names|
|-|-|other changes can be found in issues #4665 , #4666 , #4667 |

### LINKED ISSUE #4671: Main Menu: change Quality tab
mmquality

|Description of Issue|Possible Solution|  Warnings |
|:-----------------------------------------:|:-------------------------------------------------:|:---------------:|
| The three lowest items do not fit the name of the tab as they do not directly or exclusively affect quality|move those items to Tools tab| see prototype in #4672 |
|-|-|other changes can be found in issues #4665 , #4666 , #4667 |

### LINKED ISSUE #4672: Main Menu: change Tools tab
mmtools

|Description of Issue|Possible Solution|  Warnings |
|:-----------------------------------------:|:-------------------------------------------------:|:---------------:|
| Adding a sublibrary has a global effect on the library|move _New sublibrary based on AUX file_ to Library|see prototype in #4670 |
| Opening a folder, file or URL/DOI apply to only one entry, which does not match the other tools|remove items from main menu|-|
| _OpenOffice/LibreOffice connection_ and _Open Terminal here_ open a window and are not tools|move items to View tab|see prototype in #4673 |
|_Abbreviate Journal names_ needs two items for different abbreviation styles|use an additional level of hierarchy|-|
|-|-|other changes can be found in issues #4665 , #4666 , #4667 |

### LINKED ISSUE #4673: Main Menu: change View tab
mmview

|Description of Issue|Possible Solution|  Warnings |
|:-----------------------------------------:|:-------------------------------------------------:|:---------------:|
| Toggle items are named inconsistently;<br/>the toggle state is not visible in the menu|Remove _toggle_ from the item names;<br/>transform to toggle buttons | when a toggle button is used, the menu tab should not close automatically like with other menu functions, as the toggle buttons themslves hold feedback for the user|
| Opening items are named inconsistently|Use the keyword _open_ in all items that make something visible|-|
| _Select all_ is not a viewing function|remove item from main menu| make tools and other functions applicable to all entries for a long term solution|
| Changing preview styles takes up additional menu items|use additional hierarchy level| this would be an experimental change, because it would create a submenu for a toggle button|
|-|-|other changes can be found in issues #4665 , #4666 , #4667 |

### PR DESCRIPTION
**Pull request by davidemdot**
Group and reorder the Main Menu (File, Edit, Library, Quality, Tools, and View tabs & icons). Now it looks like on the @MartinKarim's prototypes.

1_file

2_edit

3_library

4_quality

5_tools

6_view

This PR fixes #4666 & fixes #4667 & fixes #4668 & fixes #4669 & fixes #4670 & fixes #4671 & fixes #4672 & fixes #4673.

----

- [x] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [x] Screenshots added in PR description (for bigger UI changes)
- [x] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/resources/l10n/JabRef_en.properties [hunk 1]
```diff
@@ -2184,3 +2176,20 @@ Keyword\:=Keyword:
 Keyword\ \"%0\"\ already\ exists=Keyword "%0" already exists
 Remove\ keyword=Remove keyword
 Are\ you\ sure\ you\ want\ to\ remove\ keyword\:\ \"%0\"?=Are you sure you want to remove keyword: "%0"?
+
+Edit\ string\ constants=Edit string constants
+Export\ all\ entries=Export all entries
+Generate\ BibTeX\ keys=Generate BibTeX keys
+Groups\ interface=Groups interface
+ISO=ISO
+MEDLINE=MEDLINE
```
Comment (Siedlerchr) [inline src/main/resources/l10n/JabRef_en.properties]: MEDLINE and ISO are both acronyms for organizations which are not really translatable, so it makes no sense to have them in here

File: src/main/java/org/jabref/gui/actions/StandardActions.java [hunk 2]
```diff
@@ -109,33 +112,33 @@
     MANAGE_CONTENT_SELECTORS(Localization.lang("Manage content selectors"), IconTheme.JabRefIcons.PREFERENCES),
     MANAGE_CITE_KEY_PATTERNS(Localization.lang("BibTeX key patterns")),
 
-    TOGGLE_PREVIEW(Localization.lang("Toggle entry preview"), IconTheme.JabRefIcons.TOGGLE_ENTRY_PREVIEW, KeyBinding.TOGGLE_ENTRY_PREVIEW),
-    EDIT_ENTRY(Localization.lang("Edit entry"), IconTheme.JabRefIcons.EDIT_ENTRY, KeyBinding.EDIT_ENTRY),
-    SHOW_PDV_VIEWER(Localization.lang("Show document viewer"), IconTheme.JabRefIcons.PDF_FILE),
-    NEXT_PREVIEW_STYLE(Localization.lang("Next preview layout"), KeyBinding.NEXT_PREVIEW_LAYOUT),
-    PREVIOUS_PREVIEW_STYLE(Localization.lang("Previous preview layout"), KeyBinding.PREVIOUS_PREVIEW_LAYOUT),
+    EDIT_ENTRY(Localization.lang("Open entry editor"), IconTheme.JabRefIcons.EDIT_ENTRY, KeyBinding.EDIT_ENTRY),
+    SHOW_PDV_VIEWER(Localization.lang("Open document viewer"), IconTheme.JabRefIcons.PDF_FILE),
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/actions/StandardActions.java]: Please rename PDV to PDF. Seems like it was a typo someone introduced.

File: src/main/java/org/jabref/gui/JabRefFrame.java [hunk 3]
```diff
@@ -788,18 +787,15 @@ private MenuBar createMenu() {
...
-
-                new SeparatorMenuItem()
-
+                factory.createMenuItem(StandardActions.MANAGE_KEYWORDS, new ManageKeywordsAction(this))
         );
 
         if (Globals.prefs.getBoolean(JabRefPreferences.SPECIALFIELDSENABLED)) {
             boolean menuItemAdded = false;
-            if (Globals.prefs.getBoolean(JabRefPreferences.SHOWCOLUMN_RANKING)) {
+            /*if (Globals.prefs.getBoolean(JabRefPreferences.SHOWCOLUMN_RANKING)) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/JabRefFrame.java]: Why is this commented out?

File: src/main/resources/l10n/JabRef_en.properties (hunk 1 above)
Comment (davidemdot) [reply src/main/resources/l10n/JabRef_en.properties]: Good point. I will do it.

File: src/main/java/org/jabref/gui/actions/StandardActions.java (hunk 2 above)
Comment (davidemdot) [reply src/main/java/org/jabref/gui/actions/StandardActions.java]: I guess it comes from PreviewDocumentViewer or something like that. Anyway, I agree that PDF is a better choice.

File: src/main/java/org/jabref/gui/JabRefFrame.java (hunk 3 above)
Comment (davidemdot) [reply src/main/java/org/jabref/gui/JabRefFrame.java]: Two people said in #4669 we will remove that. I have preferred to leave it as a comment, waiting for more opinions, but I have forgotten the `[WIP]` prefix (this is my first contribution to this project ;-)).

Do I put it back?

Comment (davidemdot) [reply src/main/java/org/jabref/gui/JabRefFrame.java]: I have put it back for the moment.

Thanks for your review!

**Comment by tobiasdiez** [comment]
Thanks a lot for your work on this. The code changes look good to me. I believe that removing the three tools "Opening a folder, file or URL/DOI apply" disables the associated shortcuts (e.g. F4 to open a file). Can you please make sure the shortcuts work as expected and if not add some code for this here:
https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/gui/JabRefFrame.java#L315 

I've a few questions for @MartinKarim, why he proposed certain changes (some of which are not too intuitive for me):
- "Replace string" from edit to tools & "Set/clear/rename/append fields" from edit to library (#4669): Both of these operations change field values of the currently selected entries (or all entries) and thus for me have the same quality as the "Manage keywords" feature (which is essentially a special version of "set fields"). Moreover, all these three features don't involve any magic on JabRef's part (in contrast to "tools") but they are different versions of bulk edits (but that's definitely a developer-viewpoint and not a user-centric one ;-) ). I'm fine with moving all three to "Tools" although I feel this menu gets overloaded a bit.
Moreover, I would suggest t rename "Replace string" by "Find and replace".

- "Find unlinked files" from quality to tools: for me this feature is connected to the "set file link" feature, because once you linked a file to an entry that it is no longer reported under "unlinked files" and vice versa the "find unliked files" dialog allows to link files to entries. So personally, I would group them together. 

- "New sublibrary based on AUX file" from tools to library with the reasoning that "Adding a sublibrary has a global effect on the library". However, this feature does not add entries to the current library but uses the AUX file to select certain entries and then creates a new library containing only these entries. So in some sense it is a certain kind of export. Thus, for me it should go to the export menu under file or stay in tools.

File: src/main/java/org/jabref/gui/JabRefFrame.java (hunk 3 above)
Comment (tobiasdiez) [reply src/main/java/org/jabref/gui/JabRefFrame.java]: I took care of this in https://github.com/JabRef/jabref/pull/4708 ;-).

**Comment by davidemdot** [comment]
Yes, you are right, now those shortcuts are disabled. Although Ctrl + A for selecting all is still working.
In addition, I just realized that sometimes the check symbol of the CheckItem objects (Tools > Web search, Groups interface...) is not synchronized with it side panel.

I will deal with it in a while!

**Comment by Siedlerchr** [comment]
The check synchronization might be a problem with setting the values correctly in the preferences, I remember having similar issues with shown/not shown in the OpenOffice Panel. I had to explicitly set the preferences to new values.

**Comment by davidemdot** [comment]
I hope the check synchronization is solved now. Also, I have recovered the previous entry preview elements and behavior.

Please, tell me if it is OK. Thank you!

**Comment by Siedlerchr** [comment]
You seem to have a compile error now. Probably because of the parameter removal in the Check menu stuff.
You can click on the Travis failed icon and will see the jobs that failed. You then can expand them and will see the errors

**Comment by davidemdot** [comment]
Yes, my fault. I am so busy these days and cannot focus on it properly. I will solve this ASAP.
Sorry for wasting your time.

**Comment by Siedlerchr** [comment]
No problem. Take your time.

**Comment by davidemdot** [comment]
@Siedlerchr What about the coverage test? Do I need to get an equal or greather percent?

**Comment by Siedlerchr** [comment]
You can ignore the code coverage and the dependency update test.
Codacy is sometimes a bit difficult, as for some FXML stuff it is generating false positives (e.g. marking functions as unused, but does not take into account that these are defined in fxml)
We are generally not so focused on code coverage but prefer tests for the most important model/logic functionality.

**Comment by davidemdot** [comment]
By the way, I have forgotten to say that this code keeps the associated shortcuts because those choices are also in the right-click menu.

**Comment by MartinKarim** [comment]
@tobiasdiez I am sorry for being late in answering your questions. Here are my thoughts on why I moved the items/features you mentioned:

1. _Replace string_: I do think that the the main difference between this feature and _Manage Keywords_ is the possibility to change a string in all entries and not just selected ones, meaning the scope of the function separates it from the other items in _Edit_.
Additionally, replacing a string is not bound to a field like _Mangage Keywords_ or _Rank entry_. Therefore, it has a more general purpose, I would argue.
However, I agree that it does not completely fit into the _Tools_ tab as well. Moving it there was a try to give the function more exposure because many users did not know about. The _Tools_ tab is a central part of the users' workflow, which is why I chose that tab.
The name change for the feature is great, I think, because it matches the name of the same feature in other programs.

2. _Set/clear/rename/append fields_: I proposed to move this feature to library because I thought that the four functions it contains had a global effect on all entries. I have noticed, that this was a misunderstanding on my part, so I will have another look at the feature.
But I still think, that the feature as a whole does not fit the _Edit_ tab because of the "rename" part. Renaming is not really a content change and it has more complex implications for an entry. If the renaming function was split from the rest, keeping it in _Edit_ seems appropriate.

3. _Find unlinked files_: While _Set file links_ does only improve the quality of existing entries by associating a file, this feature is a bit more versatile. It can also help to find files in a folder which do not even have a bibliography entry yet and can even be used to extract bibliographic information from a PDF. I think that the ways of using this feature go beyond QA measures, which is why I proposed to move it.

4. _New sublibrary based on AUX file_: I think that you are right in regard to this function being an export function and that it could be moved to _Export_ as well. Then it would have to be renamed though into something like "Export entries cited in AUX file". However, export has the connotation of some change in the data format, which is not really the case with this feature because it only creates a subset of the current bibliography. I found the _Library_ tab appropriate because the feature is bound to the active library and therefore "depends" on it (and an AUX file).

I think all four points are up to discussion and depend on the view of the individual. This is why I think that especially the grouping of items in the main menu would profit from more user data to have a more objective center in that discussion.

**Comment by tobiasdiez** [comment]
Thanks @MartinKarim for your detailed explanation! That clarifies a lot for me and I'm fine with these changes. For the sake of moving forward with this PR, I would propose to
- Rename "Replace string" to "Find and replace"
- Move "New sublibrary based on AUX file" to "Tools"
as part of this PR and then merge. @davidemdot could you please implement these changes? Thanks!

Then, as a second step, we should discuss in the devcall:
- Set/clear/rename/append fields: edit, library or tools?
- New sublibrary based on AUX file: library, tools, export?
- How to improve telemetry collection?

**Comment by davidemdot** [comment]
@tobiasdiez, I think it is done now!

**Pull request merged**